include/ada/unicode-inl.h

```diff
@@ -0,0 +1,25 @@
+/**
+ * @file unicode-inl.h
+ * @brief Definitions for unicode operations.
+ */
+#ifndef ADA_UNICODE_INL_H
+#define ADA_UNICODE_INL_H
+#include <algorithm>
+#include "ada/unicode.h"
+
+/**
+ * @namespace ada::unicode
+ * @brief Includes the declarations for unicode operations
+ */
+namespace ada::unicode {
+ada_really_inline size_t percent_encode_index(const std::string_view input,
+                                              const uint8_t character_set[]) {
+  return size_t(std::find_if(input.begin(), input.end(),
+                             [character_set](const char c) {
+                               return character_sets::bit_at(character_set, c);
+                             }) -
```

Comment:
Is this line correct? I'm confused...

Reply:
What do you find confusing about it? It seeks the location of the first matching character, as an iterator and then converts this iterator into an index. To make it clearer, I have rewritten it with std::distance.